### PR TITLE
Add support for additional currencies with formatting

### LINKED ISSUE #15: Add ability for admins to switch currency from dollars
## Details

Currently the US dollar is hard coded into the plugin code. We should provide admins the ability to change this currency to the one of their choosing.

Note: This is not a donor-facing currency switching capability. 

## Acceptance Criteria

-   [x] New currency option is added to Gutenberg listing Stripe supported currencies
-   [x] Currency displays on the frontend properly
-   [x] Currency is formatted correctly
-  [x] The donation is processed in the correct currency
-  [x] Multiple donation forms on a page respect the chosen currency for the particular form

### PR DESCRIPTION
**Pull request by DevinWalker**
Resolves #15 #38 #40 

## Description

This PR adds support for switching the currency of a donation form from USD to any other Stripe-supported currency.

I have reworked the block's panel options with a new section called "Donation Settings" that now includes the new currency select and donation levels that were previously under "Appearance Settings".

If Stripe does not support the chosen currency an error notice will appear when attempting to move from Step 1 to Step 2 (payment step).

This PR also includes currency formatting using `Intl.NumberFormat` as described in #15 

I also added support for zero decimal currencies.

## Affects

Currency formatting.

## Testing Instructions

1. Switch to zero decimal currency and test to ensure it processes correctly. The Japanese Yen is a popular one.
2. Check that the currency formatting is working correctly for various currencies and cross-reference with popular eCommerce sites using that currency. Example: Iceland.
3. Attempt to accept a donation using a non-Stripe supported currency and ensure an error notice displays.

## Other

In the future we'll likely want to trim up the list to only the supported Stripe currencies. They support the vast majority of them though.

As well, RTL languages and currencies are not very well supported. We'll have to tackle that in a separate issue.

### CODE REVIEW

**Comment by DevinWalker** [comment]
@JasonTheAdams the merge conflicts have been resolved for this PR. As well, here's a short video that should provide some context for the features in this update: https://www.loom.com/share/ca459483ecc34e82a51e00178b650634

File: src/donationForm.js [hunk 1]
```diff
@@ -86,14 +87,20 @@ const DonationForm = (props) => {
 
         setIsLoading(true);
 
+        // 💵 How much should be charged? Converts to cents for non-zero decimal currencies.
+        const chargeAmount = zeroDecimalCodes.includes(props.attributes.currencyCode) ? donationAmount : donationAmount * 100;
+
+        console.log(chargeAmount);
```
Comment (JasonTheAdams) [inline src/donationForm.js]: Oops. 😁

File: src/donationForm.js [hunk 2]
```diff
@@ -86,14 +87,20 @@ const DonationForm = (props) => {
 
         setIsLoading(true);
 
+        // 💵 How much should be charged? Converts to cents for non-zero decimal currencies.
+        const chargeAmount = zeroDecimalCodes.includes(props.attributes.currencyCode) ? donationAmount : donationAmount * 100;
```
Comment (JasonTheAdams) [inline src/donationForm.js:91]: Some currencies have 3 minor units. See my review comment for further details on how to get about this.

**Review by JasonTheAdams — APPROVED** [review]
@DevinWalker I noticed that we're storing the country code, language code, currency code, and currency symbol. Depending on what customization we want to offer, we may only need the currency code (e.g. USD).

First, the locale code is separate from the currency code. The reason is the locale is intended to reflect the format to the user in a way that makes most sense to them. So it's possible to have the currency as USD, but use a German locale. This could be locked into the currency, but I think it makes more sense to let the user viewing the donation form see it in their own format. To get the locale of the user one can use the `navigation.language` property.

Next, a useful sibling of the `Intl.NumberFormat.format` function is the [Intl.NumberFormat.formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts) function. With this function you can pass a string and get all the pieces of the formatted currency. This is useful as it provides further customization (if desired), but it also handles the number of minor digits a currency can support. For example:

```js
const amount = 25.1234;

var formatter = new Intl.NumberFormat(navigator.language, {
  style: "currency",
  currency: "USD"
});

const parts = formatter.formatToParts(amount);

const charge = parts.reduce(
  (acc, { type, value }) => ["integer", "fraction"].includes(type) ? acc + value : acc,
  ""
);
```

An interesting benefit of this is that the charge value returned is a String, not a Number. This means that the currency sent to the gateway isn't affected by floating-point issues that can affect the value sent. Notice that we're ignoring the decimal position, which means we're getting the value back in its minor unit.

In short, the Intl functions should do what we need such that we only need the currency code at least, and locale at most.

**Comment by DevinWalker** [comment]
@JasonTheAdams I didn't know about `navigation.language`, which looks like exactly what we want. Thanks for the tip, I'll refactor this now.

File: src/donationForm.js (hunk 2 above)
Comment (JasonTheAdams) [reply src/donationForm.js:91]: @DevinWalker I'm confused. Did you decide this is fine or something?

Comment (DevinWalker) [reply src/donationForm.js:91]: Yeah... It's not likely anyone is using a three decimal currency right now so I figured leave that for another day and get a release out.

Comment (JasonTheAdams) [reply src/donationForm.js:91]: Normally, I would agree, but keep in mind that if someone with a currency other than 0/2 minor units uses the block it's going to charge the wrong amount to people's accounts... that's real bad. 😬

Comment (DevinWalker) [reply src/donationForm.js:91]: I'll make sure I fix it asap. Just for you, and all the three decimal currencies out there. 🙂
